query-planner-js/src/composedSchema/toAPISchema.ts

```diff
@@ -40,6 +41,8 @@ export function toAPISchema(schema: GraphQLSchema): GraphQLSchema {
 
   schema.__apiSchema = apiSchema;
 
+  assertValidSchema(schema);
```

Comment:
Was it intentional to put this _after_ we've cached the api schema? This means that if the schema is invalid, the first call to `toAPISchema` will throw, but followup calls will succeed with a presumably broken schema. Don't know the gateway code well enough to say if it matters in practice, but I intuitively would have put the validation earlier to avoid ever using an invalid schema.

Reply:
Since `toAPISchema()` is part of the public API, I'd suggest it'd be good to validate at the start of the function. You'd also presumably want to validate at the end of the function to catch user errors like e.g. removing all the fields of an object/interface type but not removing the object/interface type itself (i.e. empty object/interface types).